src/pgduckdb_background_worker.cpp

```diff
@@ -280,9 +312,11 @@ ShmemStartup(void) {
 		 */
 		MemSet(BgwShmemStruct, 0, size);
 		SpinLockInit(&BgwShmemStruct->lock);
-		BgwShmemStruct->bgw_latch = nullptr;
-		BgwShmemStruct->activity_count = 0;
-		BgwShmemStruct->bgw_session_hint_is_reused = false;
+
+		HASHCTL info;
+		info.keysize = sizeof(Oid);
+		info.entrysize = sizeof(BgwStatePerDB);
+		BgwShmemStruct->statePerDB = ShmemInitHash("ProcBgwStatePerDB", 16, 2048, &info, HASH_ELEM | HASH_BLOBS);
```

Comment:
Curious to hear your thoughts on these defaults? (16=initial item #, 2048=maximum item #)

Reply:
I think as a maximum `max_worker_processes` should be enough for a maximum size, 2048 is a bit arbitrary (and usually excessive). Lowering it to `max_worker_processes` has one issue though currently: we don't remove entries from the hashmap when a background worker dies. I think we should do that with PG_TRY + PG_FINALLY. Also the `// Unreachable` comment is just plain wrong, we have a break in the loop now.

Reply:
`max_worker_processes` is a good idea. I also had planned to work on the unregister, but it shouldn't be in our way for the number of items here since it is indexed per database `Oid` - so if one dies there's hopefully one that will come in its place (or the slot will be wasted, but there won't be a continuous increase)